homeassistant/components/input_number.py

```diff
@@ -64,15 +74,16 @@ def _cv_input_slider(cfg):
             vol.Optional(CONF_STEP, default=1):
                 vol.All(vol.Coerce(float), vol.Range(min=1e-3)),
             vol.Optional(CONF_ICON): cv.icon,
-            vol.Optional(ATTR_UNIT_OF_MEASUREMENT): cv.string
-        }, _cv_input_slider)
+            vol.Optional(ATTR_UNIT_OF_MEASUREMENT): cv.string,
+            vol.Optional(CONF_MODE, default=MODE_SLIDER): cv.string,
```

Comment:
Instead of `cv.string` use `vol.In(MODE_BOX, MODE_SLIDER)` and remove the validation from `_cv_input_number`